src/domain/user/user.resolver.fields.ts

```diff
@@ -0,0 +1,29 @@
+import { UseGuards } from '@nestjs/common';
+import { Resolver } from '@nestjs/graphql';
+import { Parent, ResolveField } from '@nestjs/graphql';
+import { Roles } from '@utils/decorators/roles.decorator';
+import { GqlAuthGuard } from '@utils/authentication/graphql.guard';
+import { Profiling } from '@utils/logging/logging.profiling.decorator';
+import { User } from '@domain/user/user.entity';
+import { UserService } from './user.service';
+import { RestrictedGroupNames } from '@domain/user-group/user-group.entity';
+import { MemberOf } from './memberof.composite';
+
+@Resolver(() => User)
+export class UserResolverFields {
+  constructor(private userService: UserService) {}
+
+  @Roles(RestrictedGroupNames.Members)
```

Comment:
This will restrict only to the Members, not all levels. Roles need to be added explicitly.